Compilation error when Speaker list is empty

If the speaker-list of a talk is empty, we end up accessing a `nil` value during compilation:

```
bundle exec nanoc
Loading site… done
Compiling site…

Captain! We’ve been hit!

Message:

NoMethodError: undefined method `join' for nil:NilClass

Stack trace:

  0. lib/ical_helpers.rb:9:in `generate_event'
  1. lib/ical_helpers.rb:22:in `block in generate_ical_items'
  2. lib/ical_helpers.rb:21:in `each'
  3. lib/ical_helpers.rb:21:in `generate_ical_items'
  4. /home/clouduser/UnFUG/unfug.org/Rules:17:in `block in parse'
  5. /home/clouduser/.gem/ruby/2.3.0/gems/nanoc-4.8.10/lib/nanoc/rule_dsl/action_provider.rb:49:in `instance_eval'
  6. /home/clouduser/.gem/ruby/2.3.0/gems/nanoc-4.8.10/lib/nanoc/rule_dsl/action_provider.rb:49:in `block in preprocess'
  7. /home/clouduser/.gem/ruby/2.3.0/gems/nanoc-4.8.10/lib/nanoc/rule_dsl/action_provider.rb:48:in `each_value'
  8. /home/clouduser/.gem/ruby/2.3.0/gems/nanoc-4.8.10/lib/nanoc/rule_dsl/action_provider.rb:48:in `preprocess'
  9. /home/clouduser/.gem/ruby/2.3.0/gems/nanoc-4.8.10/lib/nanoc/base/services/compiler/stages/preprocess.rb:17:in `run'
  ... 37 more lines omitted. See full crash log for details.

If you believe this is a bug in Nanoc, please do report it at
-> https://github.com/nanoc/nanoc/issues/new <-

A detailed crash log has been written to ./crash.log.
make: *** [Makefile:2: all] Error 1
```

--- a/content/index.html
+++ b/content/index.html
@@ -23,7 +23,7 @@ title: Home
         <% if item.attributes[:speakers] or item.attributes[:author] %>
             <div class="frontpage_item_author">
                 <% if is_talk? item %>
-                    Speaker(s): <%= item.attributes[:speakers].join(", ") %>
+                    Speaker(s): <%= (item.attributes[:speakers] or []).join(", ") %>
                 <% else %>
                     Author: <%= item.attributes[:author] %>
                 <% end %>
@@ -68,7 +68,7 @@ title: Home
         <% if item.attributes[:speakers] or item.attributes[:author] %>
             <div class="frontpage_item_author">
                 <% if is_talk? item %>
-                    Speaker(s): <%= item.attributes[:speakers].join(", ") %>
+                    Speaker(s): <%= (item.attributes[:speakers] or []).join(", ") %>
                 <% else %>
                     Author: <%= item.attributes[:author] %>
                 <% end %>

--- a/content/talks.html
+++ b/content/talks.html
@@ -30,7 +30,7 @@ kind: :page
                         <%= link_to a[:title].to_s, item %>
                       <% end%>
                     </td>
-                    <td><%= a[:speakers].join(", ") %></td>
+                    <td><%= (a[:speakers] or []).join(", ") %></td>
                 </tr>
             </div>
         <% end %>
@@ -71,7 +71,7 @@ kind: :page
                             <%= link_to a[:title].to_s, item %>
                           <% end%>
                         </td>
-                        <td><%= a[:speakers].join(", ") %></td>
+                        <td><%= (a[:speakers] or []).join(", ") %></td>
                     </tr>
                 </div>
             <% end %>

--- a/layouts/talk.html
+++ b/layouts/talk.html
@@ -2,7 +2,7 @@
     <ul>
         <li>Titel: <%= @item.attributes[:title] %></li>
         <li>Datum: <%= @item.attributes[:date] %></li>
-        <li>Redner: <%= @item.attributes[:speakers].join(", ") %></li>
+        <li>Redner: <%= (@item.attributes[:speakers] or []).join(", ") %></li>
         <% if @item.attributes[:tags] and not @item.attributes[:tags].empty? %>
             <li>
                 Tag(s): <%= tags_for(@item, separator: ", ", base_url: "/tags/") %>

--- a/lib/ical_helpers.rb
+++ b/lib/ical_helpers.rb
@@ -6,7 +6,11 @@ def generate_event item, cfg
   event.dtstart = item[:date].to_datetime + (cfg[:page][:unfug][:start] / 24.0)
   event.dtend   = item[:date].to_datetime + (cfg[:page][:unfug][:end]   / 24.0)
   event.summary = "UnFUG"
-  event.description = item[:speakers].join(", ") + " - " + item[:title]
+  if item[:speakers]
+    event.description = item[:speakers].join(", ") + " - " + item[:title]
+  else
+    event.description = item[:title]
+  end
   event
 end
 

